packages/server/dataloader/azureDevOpsLoaders.ts

```diff
@@ -0,0 +1,301 @@
+            .get('teamMemberIntegrationAuths')
+            .load({service: 'azureDevOps', teamId, userId})
+
+          if (!azureDevOpsAuthToRefresh) {
+            console.log('error line 61')
+            return null
+          }
+          const {accessToken: existingAccessToken, refreshToken} = azureDevOpsAuthToRefresh
+          console.log(
+            `returned token in freshAzureDevOpsAuth - existingAccessToken: ${existingAccessToken} | refreshToken: ${refreshToken}`
+          )
+          const decodedToken = existingAccessToken && (decode(existingAccessToken) as any)
+          const now = new Date()
+          const inAMinute = Math.floor((now.getTime() + 60000) / 1000)
+          if (!decodedToken || decodedToken.exp < inAMinute) {
+            if (!refreshToken) {
+              return null
+            }
+            console.log('about to call refresh token - the token expired')
+            const oauthRes = await AzureDevOpsServerManager.refresh(refreshToken)
+            if (oauthRes instanceof Error) {
+              //sendToSentry(oautRes)
+              return null
+            }
+            const {accessToken, refreshToken: newRefreshToken} = oauthRes
+            const updatedRefreshToken = !!newRefreshToken ? newRefreshToken : refreshToken
+            // const updatedSameAccountAzureDevOpsAuths = userAzureDevOpsAuths
+            //   .filter((auth) => auth.accountId === azureDevOpsAuthToRefresh.accountId)
+            //   .map((auth) => ({
+            //     ...auth,
+            //     accessToken,
+            //     refreshToken: updatedRefreshToken
+            //   }))
+            // console.log(updatedSameAccountAzureDevOpsAuths)
+            // await upsertAzureDevOpsAuths(updatedSameAccountAzureDevOpsAuths)
+            const newAzureDevOpsAuth = {
+              ...azureDevOpsAuthToRefresh,
+              accessToken,
+              refreshToken: updatedRefreshToken
+            }
+            console.log(`object prior to upsert ${newAzureDevOpsAuth}`)
+            await upsertTeamMemberIntegrationAuth(newAzureDevOpsAuth)
+            return newAzureDevOpsAuth
+          }
+
+          console.log('returning azureDevOpsAuthToRefresh')
+          return azureDevOpsAuthToRefresh
+        })
+      )
+      return results.map((result) => (result.status === 'fulfilled' ? result.value : null))
+    },
+    {
+      ...parent.dataLoaderOptions,
+      cacheKeyFn: (key) => `${key.userId}:${key.teamId}`
+    }
+  )
+}
+
+export const azureDevOpsAllWorkItems = (
+  parent: RootDataLoader
+): DataLoader<TeamUserKey, WorkItem[] | undefined, string> => {
+  return new DataLoader<TeamUserKey, WorkItem[] | undefined, string>(
+    async (keys) => {
+      const results = await Promise.allSettled(
+        keys.map(async ({userId, teamId}) => {
+          const returnWorkItems = [] as WorkItem[]
+          const auth = await parent.get('freshAzureDevOpsAuth').load({teamId, userId})
+          if (!auth) return undefined
```

Comment:
+1 It's easier to handle in the rest of the code if we just return `[]` here. Not having `auth` is handled separately anyways.